reconcile/templating/renderer.py

```diff
@@ -166,12 +165,14 @@ def unpack_static_variables(
 def unpack_dynamic_variables(
     collection_variables: TemplateCollectionVariablesV1, gql: gql.GqlApi
 ) -> dict[str, dict[str, Any]]:
-    if not collection_variables.dynamic:
-        return {}
-
-    return {
-        dv.name: gql.query(dv.query) or {} for dv in collection_variables.dynamic or []
-    }
+    static = collection_variables.static or {}
+    dynamic: dict[str, dict[str, Any]] = {}
+    for dv in collection_variables.dynamic or []:
+        query = process_jinja2_template(
+            body=dv.query, vars={"static": static, "dynamic": dynamic}
+        )
+        dynamic[dv.name] = gql.query(query)
```

Comment:
pass `dynamic` into `vars` may have some unexpected result depending on processing order, and there was a default value `{}` if `query` returns no response, not sure if used anywhere, better to add test cases for this change at https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/templating/test_renderer.py

Reply:
I expect the order to be kept (is that correct?), and was doing that to be able to reference data from one dynamic variable to an other.

I'll add the missed default and check tests

Reply:
and the default `{}` was used in final result `gql.query(dv.query) or {}` instead of `dv.query`

Reply:
done in https://github.com/app-sre/qontract-reconcile/pull/4427/commits/4492a9a266cb093880b9ef6b8d259859aa4f4eb9 and https://github.com/app-sre/qontract-reconcile/pull/4427/commits/d407418349495e207ee6ed82bd6eb478b5ddba12